batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -202,12 +202,12 @@ static OpenmrsUtil createOpenmrsUtil(String sourceUrl, String sourceUser, String
 		
 		@Setup
 		public void Setup() {
-			fhirContext = FhirContext.forDstu3();
+			fhirContext = FhirContext.forR4();
 			fhirStoreUtil = FhirStoreUtil.createFhirStoreUtil(sinkPath, sinkUsername, sinkPassword,
 			    fhirContext.getRestfulClientFactory());
 			openmrsUtil = createOpenmrsUtil(sourceUrl, sourceUser, sourcePw, fhirContext);
 			fhirSearchUtil = new FhirSearchUtil(openmrsUtil);
-			parquetUtil = new ParquetUtil(fhirContext, parquetFile);
+			parquetUtil = new ParquetUtil(FhirContext.forDstu3(), parquetFile);
```

Comment:
I feel it is better to remove the fhirContext constructor argument from `ParquetUtil` and instead initialize `fhirContext` instance variable inside the constructor. I see two main benefits for passing the arg (i) for reusing context, which is an expensive object to create and (ii) mocking the context in unit-testing of `ParquetUtil`. But after your change, it seems (i) is gone. Also (ii) was never needed, so I feel it is probably simpler to drop the constructor argument. WDYT?

Reply:
yeah, I originally did it that way, but changed it back for consistency. Your approach makes sense though. I think in general we might want to refactor the code to use autowired context. @ibacher any thoughts on this? I'm not the biggest spring expert, and ran into some issues with that approach previously.

Reply:
I'm actually not a huge fan of Spring, but dependency injection and inversion of control can be useful as it kind of forces you to write code in a way that supports @bashir2's (ii) above. I don't know if a framework is really necessary here, but I might suggest taking a look at Quarkus or (for the Google points) Dagger, both of which do compile-time dependency resolution.

Reply:
Yeah, I am also very cautious of introducing Spring in this code base. It is a great framework but I feel it is too bulky for just solving dependency injection problem. I also have concerns about using it in building libraries (which might be because of limited knowledge about Spring).

I agree with @ibacher that we should consider adding DI frameworks and I have heard good things about Dagger (well, I am obviously biased). Also I agree @pmanko that we need a general refactoring to introduce DI.